### PR TITLE
Cart functionality

### PR DESCRIPTION
**Pull request by sameep-s**
Cart Functionality Complete:

- Add items to cart.
- Remove items from cart.
- Increase Quantity.
- Decrease Quantity.
- Price Details.
- Badge with number of items in Cart.

Screenshot (120)

Screenshot (121)

Screenshot (122)

### CODE REVIEW

File: src/pages/CardCart.jsx [hunk 1]
```diff
@@ -21,15 +25,15 @@ const CardCart = () => {
                             <div className="cart-sub-cont2">
                                 <div className="cart-info-qty">Quantity:</div>
                                 <div className="cart-qty-btn">
-                                    <button className="btn-cart-qty">-</button>
+                                    <button className="btn-cart-qty" onClick={() => dispatch_Cart({ type: "SUB_ITEM_QTY", payload: product })}>-</button>
                                     <input
                                         type="text"
                                         name=""
                                         id=""
```
Comment (Maianki) [inline src/pages/CardCart.jsx:32]: You can either give some values to these attributes or remove them.
Query -  You are not using the label for the input so can you explain what does this id attribute does here?

File: src/pages/Cart.jsx [hunk 2]
```diff
@@ -1,51 +1,82 @@
...
+import { useCart } from '../util_Contexts';
 import CardCart from './CardCart';
 import Navbar from './Navbar';
 
-function Cart() {
+const Cart = () => {
+
+    const { state_Cart } = useCart();
+    const cartItems = state_Cart.cart;
+    console.log(`cartItems:cart.jsx`, cartItems);
```
Comment (Maianki) [inline src/pages/Cart.jsx:11]: Remove the console from the code.

File: src/util_Contexts/products-context.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }, []);
+
+
+
+    return (
+        <>
+            <ProductsContext.Provider value={{ state_Products, dispatch }}>
+                {children}
+            </ProductsContext.Provider>
+        </>
```
Comment (Maianki) [inline src/util_Contexts/products-context.js:39]: You can remove these fragments here.

File: src/pages/Cart.jsx [hunk 4]
```diff
@@ -1,51 +1,82 @@
...
 import '../pages-css/cart.css';
+import { useCart } from '../util_Contexts';
 import CardCart from './CardCart';
 import Navbar from './Navbar';
 
-function Cart() {
+const Cart = () => {
+
+    const { state_Cart } = useCart();
+    const cartItems = state_Cart.cart;
```
Comment (harshalini) [inline src/pages/Cart.jsx:10]: You can destructure the cart here itself as const { state_Cart : { cart } } = useCart();

**Review by harshalini — APPROVED** [review]
Code is good, file structure is also nice, just add the deployed link of your site

**Review by Maianki — APPROVED** [review]
Great job Sameep. Has dropped few suggestions that you can work on.
Things I liked about your code 👍 
- Proper breaking up of large pieces of code into smaller reusable components.
- Good naming convention

File: src/pages/Cart.jsx (hunk 4 above)
Comment (sameep-s) [reply src/pages/Cart.jsx:10]: i will modify it.

File: src/pages/Cart.jsx (hunk 2 above)
Comment (sameep-s) [reply src/pages/Cart.jsx:11]: i will remove these.

File: src/util_Contexts/products-context.js (hunk 3 above)
Comment (sameep-s) [reply src/util_Contexts/products-context.js:39]: sure will do.

**Pull request merged**